### PR TITLE
Camera2d enhancements

### LINKED ISSUE #8175: Visual helpers to see Camera2D limits
I think a great improvement for Camera2Ds would be to have visual helpers in the editor's 2D view to see the limits.

Today (2.1), you have to move around your camera node to see where the limits are.
If you could  see the four corners of the limits rectangle in the 2D would be easier. And even better: if there were handles to change them directly in the 2D view!

### PR DESCRIPTION
**Pull request by TwistedTwigleg**
camera2d_enhancements
I added various enhancements for Camera2D:

- Added an option to draw Camera2D's limits
- Added an option to draw Camera2D's drag margin
- Made drawing the Camera's screen bounds optional
- Put all of these option under a new group, "Editor helpers", in Camera2D
- Made Camera2D's editor lines less transparent if is set to current
- Added function calls to _update_scroll() in various places so the editor lines are updated when the values change
- Added variables and set/get functions for toggling the editor lines.

I can remove or change anything if needed. I'm not sure on changing the transparency on the editor lines, but I liked being able to see if a camera was current or not at a glance.

There is one issue I couldn't figure out how to fix. If you scale the Camera2D node by a negative value, the limit bounds editor lines start moving.

Closes #8175

### CODE REVIEW

**Comment by puppetmaster-** [comment]
Is this also visible during play? (like collision shapes)
I like to have the camera limit visible during play for debugging or fine tuning.

**Comment by akien-mga** [comment]
Nice work! There seem to be some style issues, see the diff at https://travis-ci.org/godotengine/godot/jobs/244332332
You can run clang-format on your files to fix them, and or setup the precommit hook in `misc/hooks`. See http://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#mastering-the-pr-workflow-the-rebase for tips on rebasing a branch.

**Comment by bojidar-bg** [comment]
@puppetmaster- Because of the nature of camera limits, it might be very hard to see them while playing, unless you make a second camera which is out zoomed more than the camera which you usually use.

Since this PR is aimed at closing #8175, would you mind if you open this as a separate issue? 😃

File: scene/2d/camera_2d.cpp [hunk 1]
```diff
@@ -554,6 +614,36 @@ Node *Camera2D::get_custom_viewport() const {
 	return custom_viewport;
 }
 
+void Camera2D::set_screen_drawing_enabled(bool enable)
+{
+	screen_drawing_enabled = enable;
+	_update_scroll();
```
Comment (bojidar-bg) [inline scene/2d/camera_2d.cpp]: I think here (and the next two setters as well) you can use just `update()` (from CanvasItem), to force a redraw, no reason to update the position as well.

Not sure how much this applies to various other `_update_scroll()` calls above.

File: scene/2d/camera_2d.cpp [hunk 2]
```diff
@@ -643,6 +742,11 @@ void Camera2D::_bind_methods() {
 	ADD_PROPERTYI(PropertyInfo(Variant::REAL, "drag_margin_right", PROPERTY_HINT_RANGE, "0,1,0.01"), "set_drag_margin", "get_drag_margin", MARGIN_RIGHT);
 	ADD_PROPERTYI(PropertyInfo(Variant::REAL, "drag_margin_bottom", PROPERTY_HINT_RANGE, "0,1,0.01"), "set_drag_margin", "get_drag_margin", MARGIN_BOTTOM);
 
+	ADD_GROUP("Editor helpers", "editor_helper_");
```
Comment (bojidar-bg) [inline scene/2d/camera_2d.cpp]: If you want, you can shorten the code name of the group to `editor_`.

**Comment by puppetmaster-** [comment]
@bojidar-bg It's ok, I put my wish in a separate issue!

**Comment by TwistedTwigleg** [comment]
Thanks everyone! I'll start working on the changes soon, probably this weekend.

**Comment by TwistedTwigleg** [comment]
I made changes based on Puppetmaster-'s review, formatted with clang-format, and rebased.

@puppetmaster- @akien-mga  Do you guys mind checking the changes?

**Comment by bojidar-bg** [comment]
@TwistedTwigleg I think you messed up git though...

**Comment by TwistedTwigleg** [comment]
@puppetmaster- Oh, whoops! I'm still trying to get used to git. How should I fix it? Should I open a new pull request, with the new code, and close this one?

**Comment by TwistedTwigleg** [comment]
I tried to fix it, but looks like it didn't work. I'm going to make a new pull request and close this one, if that is okay with everyone.

**Comment by TwistedTwigleg** [comment]
Okay, made a new pull request, #9296. Closing this one...

**Pull request closed without merging**